src/pybind/mgr/smb/handler.py

```diff
@@ -1039,11 +1043,16 @@ def _generate_share(
         share.cephfs.subvolume,
         share.cephfs.path,
     )
+    ceph_vfs = {
+        CephFSStorageProvider.SAMBA_VFS: 'ceph_new',
+        CephFSStorageProvider.SAMBA_VFS_CLASSIC: 'ceph',
+        CephFSStorageProvider.SAMBA_VFS_NEW: 'ceph_new',
+    }[share.checked_cephfs.provider]
     cfg = {
         # smb.conf options
         'options': {
             'path': path,
-            "vfs objects": "acl_xattr ceph",
+            "vfs objects": f"acl_xattr {ceph_vfs}",
             'acl_xattr:security_acl_name': 'user.NTACL',
             'ceph:config_file': '/etc/ceph/ceph.conf',
             'ceph:filesystem': share.cephfs.volume,
```

Comment:
With [!3751](https://gitlab.com/samba-team/samba/-/merge_requests/3751) merged you may now change the fixed `ceph` prefix.

Reply:
thanks for the heads up